### ISSUE TITLE
"waiting for git blame response" flashes very quickly whenever editing a file

### ISSUE THREAD

**Issue by Japsert** [Sertion/vscode-gitblame#163]
I have the extension in my status bar. However, whenever I'm editing a file and the editor autosaves, the text "waiting for git blame response"  pops up and disappears immediately, triggered by the autosave. This is distracting to me and bothers me more than it should haha, so I would like to suggest that this text is only shown if the git blame response takes longer than ~500ms? Thank you. :)

**Comment by Sertion** [comment]
Hey Jasper! Thanks for the feature request.

I fear that it would need to be implemented the other way around: Forcing a minimum time the "processing"-message is displayed.

Would that still be interesting?

**Comment by bpasero** [comment]
@Sertion I share the same issue, wouldnt the easiest be to provide a setting for defining for how this message should appear:



![Image](https://github.com/Sertion/vscode-gitblame/assets/900690/6be4c06f-438a-4f22-a0f6-2d37b57c660f)


And then I could just put in an empty string to hide it?

**Comment by Sertion** [comment]
Thanks for the input Benjamin! Sadly hiding the status bar text will not mitigate the blinking issue. It can even make it worse as when an element gets more narrow other information that was previously hidden in the status bar gets shown.

This is why I ask about setting a minimum time shown value instead. This has a few side effects:
* The user still gets feedback on what is going on
* The blinking will be mitigated (and can be regulated by the user (if it becomes a setting))
* The chance that other status bar messages pop in and out is also mitigated

I've not yet implemented this as I do not exactly know when the implemented features is resolving the issue.

**Comment by bpasero** [comment]
Hm, so today there is an immediate progress indicator in the status bar as soon as I make changes to a line:

![Recording 2024-02-01 at 15 45 06](https://github.com/Sertion/vscode-gitblame/assets/900690/567d604d-709e-4a51-aa98-3c4df3d04102)

I am just asking for a setting to not show this in-progress message at all. Simply leave the last blame info you had at the line and transition to the "Changed by you" when you figure out it was changed by me.

So in other words, I am just asking for not showing the in-progress animation which is also spinning.

**Comment by Sertion** [comment]
I've implemented something based on the discussion here and made a preview version: [gitblame-10.9.0-preview.zip](https://github.com/Sertion/vscode-gitblame/files/14141343/gitblame-10.9.0-preview.zip).

It moves the "Waiting for git blame response"-message to the tool tip and removes the spinning from the icon. It has the issue of _other status bar items_ now being able to pop in and out, but I don't know how common that is. Please, if you have time, have a look and report back!
![image](https://github.com/Sertion/vscode-gitblame/assets/288119/e6ad4cae-2534-47b5-b0b9-6aa6cfb6ea9a)

**Comment by bpasero** [comment]
Yeah that feels already a lot better to me, thanks! I personally still do not mind to not be notified that the info is computed because its obvious when the label is not there to me at least.
